include/bm/bm_sim/headers.h

```diff
@@ -320,6 +351,88 @@ class Header : public NamedP4Object {
 #endif
 };
 
+
+// This is a helper class.
+// For now, this class is only used to represent the list, tuple and
+// structure with headers required for log_msg.
+class List : public DataRepresentation {
```

Comment:
is the order not respected since multiple vectors are being used?

Reply:
If you mean the order of the list fields, it is respected when logging message.